cardano-cli/src/Cardano/CLI/Legacy/Run/Transaction.hs

```diff
@@ -547,13 +569,11 @@ runTxBuild
           & onLeft (left . ShelleyTxCmdQueryConvenienceError . AcqFailure)
           & onLeft (left . ShelleyTxCmdQueryConvenienceError)
 
-      pp <- hoistEither . first ShelleyTxCmdProtocolParamsConverstionError $ toLedgerPParams sbe pparams
-
-      validatedPParams <- hoistEither $ first ShelleyTxCmdProtocolParametersValidationError
-                                      $ validateProtocolParameters era (Just pparams)
+      Refl <- testEquality era nodeEra
```

Comment:
since `LedgerPParams` are now parameterized by era, building `TxBodyContent` requires `era ~ nodeEra`

Reply:
We have to cast the era to the current node era otherwise we lose backwards compatibility (e.g we can't construct valid babbage transactions if our node is in the conway era). Follow up with @newhoggy about casting the era. He has a commit that implements this casting.

Reply:
Do we need backwards compatibility? From the following discussion, this is something which is hard to support: https://input-output-rnd.slack.com/archives/C057E9CJTPH/p1693494111115739

Reply:
We've decided to not support this anymore.